### DISCUSSION TITLE
Viewing HTML source

### DISCUSSION THREAD

**Discussion by ajtruckle** [MicrosoftEdge/WebView2Feedback#4508] [Q&A]
I know that we can use the built-in View Page Source mechanism. 

But, when using a XML / XSL transformation, View Page Source will only show the XML. 

I have added my own shortcuts to directly display the XML / XSL files. 

Now I want to offer to display the HTML. So View Page Source is not enough. Yet, I can right click and use Inspect and that display does have the final HTML. 

Task:
When we have a XML / XSL transformation displayed, how can we then display the final HTML in our own popup windows?

Win32

Is this possible?

**Comment by ajtruckle** [comment]
Not an answer. But I have no way forward. Please turn this into a request again. @champnic

**Reply by victorhuangwq** [reply]
@Master-Ukulele do you know if this possible? From what I know we don't have an API specifically for it.

**Reply by Master-Ukulele** [reply]
We can't modify the build-in View Page Source function. This way won't work. However, we have the `ContextMenuRequested` might be useful for your case that add your own shortcuts. cc @yildirimcagri-msft please confirm if this works.

**Reply by ajtruckle** [reply]
@Master-Ukulele I already have a custom context menu handler operational. 

I am aware that I can do my own transform on the XSL to HTML and view that as the source. But, it seems a waste of resources and code when Edge / WebView2 that already do the same thing via the Inspect mechism.

Besides, as I have previously mentioned on other tickets (which seem to be stagnant) there is no way for us to forward hotkeys to WebView2. It is fine when the browser has focus. But when our dialog has focus it eats up the hotkey. So the whole concept is flawed. Whilst WV2 lets us intercept hotkeys and redirect them back to our parent, there is no way for the parent to push them through to WV2 for processing.

I know it is a different subject but it is tied into this issue because my own context menu handler has view page source. And I have to use SendInput. And that only works if the browser has focus. If the jhotkey is called on my dialoh, even though I capture it, I can't use SendInput to push it to browser to process.

Two issues for discussion here. :)

**Reply by Master-Ukulele** [reply]
Hi @ajtruckle, let me try to answer your first topic.
> it seems a waste of resources and code when Edge / WebView2 that already do the same thing via the Inspect mechism.

Do you mean there's already the functionality in the DevTools inspect mode, so you can reuse them? I feel it's challenging in this way. It's hard to access this part of code or functionality externally from DevTools. As I know, we don't have an API for this access. I recommend you have your own transformation function; less dependency, you can have the full control of it, and probably better performance.

**Reply by Master-Ukulele** [reply]
> when our dialog has focus it eats up the hotkey

Is your dialog a modal dialog?

**Reply by ajtruckle** [reply]
@Master-Ukulele Yes, it is a modal dialog.

**Reply by Master-Ukulele** [reply]
I think that's the characteristics of modal dialog, when it's been focused, you cannot have other interaction outside of the modal dialog, including the hotkey features. If that's your own customized dialog, you might try make it not as the modal dialog.

**Reply by ajtruckle** [reply]
@Master-Ukulele  Not really. The browser control is a child control on my dialog. So I should be able to feed a child control a hotkey if required. This is nt a model window on top of a browser.
